addons/mail/static/src/js/text_emojis_field.js

```diff
@@ -122,7 +125,18 @@ var FieldTextEmojis = basicFields.FieldText.extend(MailEmojisMixin, {
     }
 };
 
+/**
+ * Extension of the FieldText that will add emojis support
+ */
+var FieldTextEmojis = basicFields.FieldText.extend(MailEmojisMixin, FieldEmojiCommon);
+
+/**
+ * Extension of the FieldChar that will add emojis support
+ */
+var FieldCharEmojis = basicFields.FieldChar.extend(MailEmojisMixin, FieldEmojiCommon);
+
 registry.add('text_emojis', FieldTextEmojis);
+registry.add('char_emojis', FieldCharEmojis);
 
 return FieldTextEmojis;
```

Comment:
You return only one of the two fields.
You need to return the 2 fields so that they can be properly overridden.

Also, you should rename your file to `mail_emojis_fields.js` since you add multiple fields.